internal/services/deviceregistry/device_registry_asset_resource.go

```diff
@@ -0,0 +1,742 @@
+		"tags":                commonschema.Tags(),
+		"extended_location_name": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"extended_location_type": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"enabled": {
+			Type:     pluginsdk.TypeBool,
+			Optional: true,
+		},
+		"external_asset_id": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"display_name": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"description": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"asset_endpoint_profile_ref": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ValidateFunc: validation.StringIsNotEmpty,
+		},
+		"manufacturer": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"manufacturer_uri": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"model": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"product_code": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"hardware_revision": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"software_revision": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"documentation_uri": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"serial_number": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"attributes": {
+			Type:     pluginsdk.TypeMap,
+			Optional: true,
+			Elem:     &pluginsdk.Schema{Type: pluginsdk.TypeString},
+		},
+		"discovered_asset_refs": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Schema{
+				Type: pluginsdk.TypeString,
+			},
+		},
+		"default_datasets_configuration": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"default_events_configuration": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"default_topic_path": {
+			Type:     pluginsdk.TypeString,
+			Optional: true,
+		},
+		"default_topic_retain": {
+			Type:         pluginsdk.TypeString,
+			Optional:     true,
+			ValidateFunc: validation.StringInSlice(assets.PossibleValuesForTopicRetainType(), false),
+			RequiredWith: []string{"default_topic_path"},
+		},
+		"datasets": {
+			Type:     pluginsdk.TypeList,
+			Optional: true,
+			Elem: &pluginsdk.Resource{
+				Schema: map[string]*pluginsdk.Schema{
+					"name": {
+						Type:     pluginsdk.TypeString,
+						Required: true,
+					},
+					"dataset_configuration": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+					"topic_path": {
+						Type:     pluginsdk.TypeString,
+						Optional: true,
+					},
+					"topic_retain": {
+						Type:         pluginsdk.TypeString,
+						Optional:     true,
+						ValidateFunc: validation.StringInSlice(assets.PossibleValuesForTopicRetainType(), false),
+					},
+					"data_points": {
```

Comment:
This should be singular

```suggestion
					"data_point": {
```